addon/webextension/background/main.js

```diff
@@ -30,10 +30,14 @@ window.main = (function () {
 
   chrome.browserAction.onClicked.addListener(function(tab) {
     if(tab.url.match(/about:(newtab|blank)/i)) {
-      sendEvent("goto-myshots", "about-newtab");
+      catcher.watchPromise(analytics.refreshTelemetryPref().then(() => {
```

Comment:
Rather than constantly re-check the prefs, you should just have the bootstrap code check once, at startup, then use a [preferences observer](https://developer.mozilla.org/en-US/Add-ons/Code_snippets/Preferences#check_For_Existence) the rest of the time.

The bootstrap code will have to send over update messages when the pref changes, a bit different than the approach you've taken here, where the webextension initiates the exchange. You can probably default telemetry to disabled, dropping packets if the initial pref message hasn't arrived before the user starts clicking.